backend/discount/delivery/discount_handler.go

```diff
@@ -0,0 +1,104 @@
+package delivery
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/PUArallelepiped/PUN-street-Universal-Access/domain"
+	"github.com/PUArallelepiped/PUN-street-Universal-Access/swagger"
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+)
+
+type DiscountHandler struct {
+	DiscountUsecase domain.DiscountUsecase
+}
+
+func NewDiscountHandler(e *gin.Engine, discountUsecase domain.DiscountUsecase) {
+	handler := &DiscountHandler{
+		DiscountUsecase: discountUsecase,
+	}
+	e.GET("/api/v1/store/:storeID/discounts", handler.GetAllDiscountByStoreId)
+	e.POST("/api/v1/store/:storeID/seasoning-discount", handler.AddSeasoningDiscount)
+	e.POST("/api/v1/store/:storeID/shipping-discount", handler.AddShippingDiscount)
+	e.POST("/api/v1/store/:storeID/event-discount", handler.AddEventDiscount)
```

Comment:
router := gin.Default()

	// Simple group: v1
	v1 := router.Group("/v1")
	{
		v1.POST("/login", loginEndpoint)
		v1.POST("/submit", submitEndpoint)
		v1.POST("/read", readEndpoint)
	}

Reply:
why no do this

Reply:
fix